sdks/python/apache_beam/typehints/typehints.py

```diff
@@ -355,9 +355,12 @@ def is_typing_generic(type_param):
 
   Such objects are considered valid type parameters.
 
-  Always returns false for Python versions below 3.7.
+  Always returns false for Python versions below 3.7. For versions 3.9 and
+  above, also permits types.GenericAlias.
   """
-  if hasattr(typing, '_GenericAlias'):
+  if hasattr(typing, '_GenericAlias') and hasattr(types, "GenericAlias"):
+    return isinstance(type_param, (typing._GenericAlias, types.GenericAlias))
```

Comment:
Would this be more clear (assuming it's still correct and accurate):
```
  if hasattr(typing, '_GenericAlias') and isinstance(type_param, typing._GenericAlias):
    # Python 3.9, 3.10:
    return True
  if hasattr(types, 'GenericAlias') and isinstance(type_param, types.GenericAlias):
    # Python 3.11+
    return True
  return False
```

Reply:
I realized the check for typing._GenericAlias is for python versions pre-3.7 that we no longer support, so I could strip that check out and make the function simpler overall